maat-orchestration/src/main/java/uk/gov/justice/laa/crime/orchestration/service/orchestration/EvidenceOrchestrationService.java

```diff
@@ -20,17 +27,48 @@ public class EvidenceOrchestrationService {
     private final RepOrderService repOrderService;
     private final UserMapper userMapper;
     private final WorkflowPreProcessorService workflowPreProcessorService;
+    private final ContributionService contributionService;
 
     public ApplicationDTO updateIncomeEvidence(WorkflowRequest workflowRequest) {
 
-        preProcessRequest(workflowRequest);
-
         RepOrderDTO repOrderDTO = repOrderService.getRepOrder(workflowRequest);
-        return incomeEvidenceService.updateEvidence(workflowRequest, repOrderDTO);
+        preProcessRequest(workflowRequest, repOrderDTO);
+
+        ApplicationDTO applicationDTO = incomeEvidenceService.updateEvidence(workflowRequest, repOrderDTO);
+
+        boolean isUpliftChanged = hasUpliftChanged(applicationDTO, repOrderDTO);
+        if (isUpliftChanged) {
+            workflowRequest.setApplicationDTO(applicationDTO);
+            return contributionService.calculate(workflowRequest);
+        } else {
+            return applicationDTO;
+        }
+    }
+
+    private static boolean hasUpliftChanged(ApplicationDTO applicationDTO, RepOrderDTO repOrderDTO) {
+        LocalDateTime currentDate = LocalDateTime.now();
+        Date today = DateUtil.toDate(currentDate);
+        var incomeEvidenceSummaryDTO = applicationDTO.getAssessmentDTO().getFinancialAssessmentDTO().getIncomeEvidence();
+        Date upliftAppliedDate = Objects.requireNonNullElse(incomeEvidenceSummaryDTO.getUpliftAppliedDate(), today);
+        Date upliftRemovedDate = Objects.requireNonNullElse(incomeEvidenceSummaryDTO.getUpliftRemovedDate(), today);
+
+        Integer financialAssessmentId = applicationDTO.getAssessmentDTO().getFinancialAssessmentDTO().getId().intValue();
+        FinancialAssessmentDTO financialAssessmentDTO = repOrderDTO.getFinancialAssessments()
+                .stream()
+                .filter(assessment -> assessment.getId().equals(financialAssessmentId))
+                .findFirst().orElse(FinancialAssessmentDTO.builder().build());
+        Date oldUpliftAppliedDate = DateUtil
+                .toDate(Objects.requireNonNullElse(financialAssessmentDTO.getIncomeUpliftApplyDate(), currentDate));
+        Date oldUpliftRemovedDate = DateUtil
+                .toDate(Objects.requireNonNullElse(financialAssessmentDTO.getIncomeUpliftRemoveDate(), currentDate));
+
+        return !upliftAppliedDate.equals(oldUpliftAppliedDate)
+                || !upliftRemovedDate.equals(oldUpliftRemovedDate);
```

Comment:
Might be an issue here if the date the uplift applied was todays date but there hasnt been any previously uplift so the default for the old date will also be todays date and hence we wont consider it an update.

Reply:
Good point! Updated the logic to use a default date instead of current date. Thanks!